README.md

```diff
@@ -0,0 +1,526 @@
+-->
+
+## Overview
+
+S7 Micro Service - device service for connecting Siemens S7(S7-200, S7-300, S7-400, S7-1200, S7-1500) devices by `ISO-on-TCP` to EdgeX.
+
+## Features
+
+- Single Read and Write
+- Multiple Read and Write
+- High performance, more 2000 items per second(depends on S7 model)
+  - Use `S7-Device01` sample device configuration, `interval` should less than `IdelTimeout`
+  - S7-1200 and S7-1500 preferred
+  - Create multiple connections to one S7 device use different device name
+
+## Prerequisites
+
+- A Siemens S7 series device with network interface
+- Enable ISO-on-TCP connection
+
+### Install and Deploy
+
+Clone and Build
+
+```shell
+git clone https://github.com/edgexfoundy-holding/device-s7.git
+cd device-s7
+make build
+```
+
+To start the device service:
+
+```shell
+export EDGEX_SECURITY_SECRET_STORE=false
+make run
+```
+
+To rebuild after making changes to source:
+
+```shell
+make clean
+make build
+```
+
+## Packaging
+
+This component is packaged as Docker image.
+
+For docker, please refer to the `Dockerfile`.
+
+### Build Docker image
+
+```shell
+make docker
+```
+
+The docker image looks like:
+
+```
+edgexfoundry/device-s7    3.1.0-dev    c84a77f45860   5 minutes ago   33.6MB
+```
+
+### Docker compose file
+
+Add to your docker-compose.yml.
+
+```yaml
+device-s7:
+  container_name: edgex-device-s7
+  depends_on:
+    consul:
+      condition: service_started
+      required: true
+    core-data:
+      condition: service_started
+      required: true
+    core-metadata:
+      condition: service_started
+      required: true
+  environment:
+    EDGEX_SECURITY_SECRET_STORE: 'false'
+    SERVICE_HOST: edgex-device-s7
+  hostname: edgex-device-s7
+  image: edgexfoundry/device-s7:0.0.0-dev
+  networks:
+    edgex-network: null
+  ports:
+    - mode: ingress
+      host_ip: 127.0.0.1
+      target: 59994
+      published: '59994'
+      protocol: tcp
+  read_only: true
+  restart: always
+  security_opt:
+    - no-new-privileges:true
+  user: 2002:2001
+  volumes:
+    - type: bind
+      source: /etc/localtime
+      target: /etc/localtime
+      read_only: true
+      bind:
+        create_host_path: true
+```
```

Comment:
This section needs to be consistent with README for other Device Services.
```suggestion
## Build Instructions

1. Clone the device-rest-go repo with the following command:

        git clone https://github.com/edgexfoundry/device-s7.git

2. Build a docker image by using the following command:

        make docker

3. Alternatively the device service can be built natively:

        make build

## Build with NATS Messaging
Currently, the NATS Messaging capability (NATS MessageBus) is opt-in at build time.
This means that the published Docker images do not include the NATS messaging capability.

The following make commands will build the local binary or local Docker image with NATS messaging
capability included.
```makefile
make build-nats
make docker-nats
```
 
## Packaging

This component is packaged as docker images.
Please refer to the [Dockerfile] and [Docker Compose Builder] scripts.

[Dockerfile]: Dockerfile
[Docker Compose Builder]: https://github.com/edgexfoundry/edgex-compose/tree/main/compose-builder
```